### PR TITLE
Builder endpoint

### PR DESCRIPTION
**Pull request by guilhermebr**

### CODE REVIEW

File: tsuru/client/build.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+		c.fs = c.GuessingCommand.Flags()
+		tag := "The image tag"
+		c.fs.StringVar(&c.tag, "tag", "", tag)
+		c.fs.StringVar(&c.tag, "t", "", tag)
+	}
+	return c.fs
+}
+
+func (c *AppBuild) Info() *cmd.Info {
+	desc := `Builds a tsuru app image. Some examples of calls are:
```
Comment (ggarnier) [inline tsuru/client/build.go]: Maybe you could mention here that this command respects `.tsuruignore` file

File: tsuru/client/build.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+		Name:    "app-build",
+		Usage:   "app-build [-a/--app <appname>] [-t/--tag <image_tag>] <file-or-dir-1> [file-or-dir-2] ... [file-or-dir-n]",
+		Desc:    desc,
+		MinArgs: 0,
+	}
+}
+
+func (c *AppBuild) Run(context *cmd.Context, client *cmd.Client) error {
+	context.RawOutput()
+	if c.tag == "" {
```
Comment (ggarnier) [inline tsuru/client/build.go]: Shouldn't you validate the tag (like checking for invalid characters)? Or do you leave this validation to the server?

File: tsuru/client/build.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	_, err = io.Copy(&respBody, resp.Body)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(safeStdout, buf.String())
+	if strings.HasSuffix(buf.String(), "OK\n") {
```
Comment (ggarnier) [inline tsuru/client/build.go]: Can't you check the response status code instead of the body?

File: tsuru/client/build.go (hunk 2 above)
Comment (guilhermebr) [reply tsuru/client/build.go]: I will do this on server side

**Pull request merged**